### PR TITLE
[docs]:Update installation steps for ballerine

### PR DESCRIPTION
**Pull request by pratapalakshmi**
## **Type**
documentation, enhancement

___

## **Description**
- Reorganized documentation sidebar to better separate installation and deployment guides.
- Introduced new deployment guides for Docker, Ansible, Kubernetes, and HTTPS configuration.
- Detailed steps for deploying Ballerine using Ansible, including inventory setup and playbook execution.
- Added a guide for configuring Ballerine to run over HTTPS with Ansible.
- Updated Kubernetes deployment guide to focus on Helm and removed Docker Compose deployment steps.
- Created a new guide for local deployment using Docker Compose.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>astro.config.mjs</strong><dd><code>Reorganization of Documentation Sidebar and Addition of Deployment </code><br><code>Guides</code></dd></summary>
<hr>

websites/docs/astro.config.mjs
<li>Reorganized the documentation sidebar, separating <code>Installation</code> and <br><code>Deployment</code> into distinct sections.<br> <li> Added new deployment guides for Docker, Ansible, Kubernetes, and <br>configuring Ballerine on HTTPS.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2177/files#diff-248df74f215a5949ecdbdcd9a8657447c7ad003e692e9d6fd19fbddfc3e5a22a">+26/-3</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ansible_deployment.mdx</strong><dd><code>New Ansible Deployment Guide for Ballerine</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

websites/docs/src/content/docs/en/getting_started/ansible_deployment.mdx
<li>Introduced a new guide for deploying Ballerine using Ansible.<br> <li> Detailed steps from installing Ansible to running the Ballerine <br>playbook.<br> <li> Provided instructions for setting up an inventory file and executing <br>the playbook.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2177/files#diff-9f417a97c537c03b7d6ba04d0aef22e0582fc0e64d2c2e00f9629e88e9ad41d9">+65/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ballerine_on_https.mdx</strong><dd><code>Guide for Running Ballerine on HTTPS</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

websites/docs/src/content/docs/en/getting_started/ballerine_on_https.mdx
<li>New guide on configuring Ballerine to run over HTTPS.<br> <li> Outlined prerequisites and detailed the setup process using Ansible.<br> <li> Described configuration variables necessary for HTTPS deployment.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2177/files#diff-50735cd56f53417f4298a6855de5980e8baa8b071b02f46397762e32575e8ea2">+54/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>kubernetes_deployment.mdx</strong><dd><code>Updated Kubernetes Deployment Guide</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

websites/docs/src/content/docs/en/getting_started/kubernetes_deployment.mdx
<li>Updated the Kubernetes deployment guide title and description for <br>clarity.<br> <li> Removed Docker Compose deployment steps, focusing on Kubernetes.<br> <li> Emphasized the use of Helm for deploying Ballerine on Kubernetes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2177/files#diff-da700a7233ded555581b013c35d8790b3dc8198223ca9b57d597d27dd44953ea">+5/-18</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>local_deployment.mdx</strong><dd><code>New Guide for Local Deployment Using Docker Compose</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

websites/docs/src/content/docs/en/getting_started/local_deployment.mdx
<li>Added a new guide for local deployment using Docker Compose.<br> <li> Provided steps for cloning the repository, switching branches, and <br>running Docker Compose.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2177/files#diff-9e780a1dd00b8ba41d25f760755e855873f9966005aacfc55823ea5f01252e55">+19/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

## Summary by CodeRabbit

- **Documentation**
	- Added a comprehensive guide on deploying the Ballerina stack using Ansible.
	- Introduced a detailed guide for configuring Ballerina to run on HTTPS.
	- Added documentation on deploying applications using Docker Compose for the Ballerina project.

### CODE REVIEW

File: websites/docs/src/content/docs/en/getting_started/installation.mdx [hunk 1]
```diff
@@ -17,7 +17,7 @@ Before you can install and use Ballerine, you will need the following:
    - Latest PNPM version (Minimum version 8.0) ([Install PNPM](https://pnpm.io/installation))
    - Docker and docker compose ([Docker](https://docs.docker.com/desktop), [Docker Compose](https://docs.docker.com/compose/install))
 
-### Installation Steps
+### Local Devlopment Steps
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/installation.mdx]: This is a typo. 🙂 
```suggestion
### Local Development Steps
```

File: websites/docs/src/content/docs/en/getting_started/installation.mdx [hunk 2]
```diff
@@ -41,6 +54,69 @@ Before you can install and use Ballerine, you will need the following:
...
+You can run the ansible playbook with the following command
+
+<CodeBlock lang={`shell`}  code={`cd ballerine/deploy/ansible/ballerine_playbook; ansible-playbook -i inventory.txt ballerine-playbook.yml`}/>
+
+The command above will use the host information from the `inventory` file.
+
+After performing these steps,
+
+The backoffice should be running on https://<backoffice_url>
+The collection flow on https://kyb_url and
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/installation.mdx]: Should be consistent with the other urls.
```suggestion
The collection flow on https://<kyb_url> and
```

File: websites/docs/src/content/docs/en/getting_started/installation.mdx [hunk 3]
```diff
@@ -41,6 +54,69 @@ Before you can install and use Ballerine, you will need the following:
...
+- `workflow_svc_url`: URL you wish to deploy Workflows-Service on
+
+Once you complete setup config vars for your app then we are ready to deploy our app on your server.
+
+8. **Run the Ansible playbook**:
+
+After complete the above step. Now the only remain step we need to do is run the ansible playbook.
+You can run the ansible playbook with the following command
+
+<CodeBlock lang={`shell`}  code={`cd ballerine/deploy/ansible/ballerine_playbook; ansible-playbook -i inventory.txt ballerine-playbook.yml`}/>
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/installation.mdx]: Most of our non-React files in the monorepo are kebab-case, don't think we should be out of the convention unless its some Ansible specific convention.
```suggestion
<CodeBlock lang={`shell`}  code={`cd ballerine/deploy/ansible/ballerine-playbook; ansible-playbook -i inventory.txt ballerine-playbook.yml`}/>
```

File: websites/docs/src/content/docs/en/getting_started/installation.mdx [hunk 4]
```diff
@@ -41,6 +54,69 @@ Before you can install and use Ballerine, you will need the following:
...
+After you completed the above step then we're pretty much done with the inventory
+
+7. **Setup your configuration vars for Ballerine**:
+
+The next step is to setup necessary configuration for your app to run such as environment variable, domain name, etc.
+
+First you need to open `deploy/ansible/ballerine_playbook/roles/setup-ballerine/defaults/main.yml` file with your editor.
+There are some variables that will need input from you to get the application start correctly
+
+- `install_dir`: The absolute path of your app's installation folder on the server (required). Default: `/home/ubuntu/ballerine`
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/installation.mdx]: Is the default `ubuntu` or the user in `/home/ubuntu/ballerine`?

File: websites/docs/src/content/docs/en/getting_started/installation.mdx (hunk 3 above)
Comment (pratapalakshmi) [reply websites/docs/src/content/docs/en/getting_started/installation.mdx]: This is more of ansible convention. So, went with it. Also change in this means change in actual code now

**Comment by Omri-Levy** [comment]
@pratapalakshmi The docs can have a deployment section or just a page “deploy with Ansible” or similar. The main installation page can always link to those sections. The installation page should have the least amount of content to get the main and basic environment going.

File: websites/docs/src/content/docs/en/getting_started/installation.mdx (hunk 4 above)
Comment (pratapalakshmi) [reply websites/docs/src/content/docs/en/getting_started/installation.mdx]: Considering a linux machine we have gone by that. But users can changes it and we have mentioned it in the above line 
[here](https://github.com/ballerine-io/ballerine/pull/2177/files#diff-59d650ebcaf91827daff10112863f6671bdf436e809d47b2fc3ce8a3d7c7d44bR94)

**Comment by pratapalakshmi** [comment]
> @pratapalakshmi The docs can have a deployment section or just a page “deploy with Ansible” or similar. The main installation page can always link to those sections. The installation page should have the least amount of content to get the main and basic environment going.

Yes I totally agree with you . But, to  setup on https and CORS. these were the basic steps . Required. I am totally in for collaboration on reducing the number of steps

File: websites/docs/astro.config.mjs [hunk 5]
```diff
@@ -47,6 +47,10 @@ export default defineConfig({
               label: `Deployment`,
               link: `/en/getting_started/deployment`,
             },
+            {
+              label: `Ansible Deployment`,
+              link: `/en/getting_started/ansibledeployment`,
```
Comment (Omri-Levy) [inline websites/docs/astro.config.mjs]:
```suggestion
              link: `/en/getting_started/ansible_deployment`,
```

File: websites/docs/src/content/docs/en/getting_started/ansibledeployment.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+---
+title: Deploy Ballerine on HTTPS using ansible
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/ansibledeployment.mdx]: Make sure casing is consistent. 🙂 
```suggestion
title: Deploy Ballerine on HTTPS using Ansible
```

File: websites/docs/src/content/docs/en/getting_started/ansibledeployment.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+description: This guide provides a step-by-step process for setting up and running the Ballerine stack with HTTPS on a remote instance.
+---
+
+import PackageManagersTabs
+  from '../../../../components/PackageManagersTabs/PackageManagersTabs.astro';
+import CodeBlock from '../../../../components/CodeBlock/CodeBlock.astro';
+
+### Installation steps on a remote virtual machine
+
+We recommend installation using ansible
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/ansibledeployment.mdx]: When referring to Ansible as a company or service let's please capitalize. 🙏🏼 
```suggestion
We recommend installation using Ansible
```

File: websites/docs/src/content/docs/en/getting_started/installation.mdx [hunk 8]
```diff
@@ -33,18 +33,23 @@ Before you can install and use Ballerine, you will need the following:
 <PackageManagersTabs code={`run monorepo:init`}/>
 
 
-## Run one of the examples, or follow a guide from the [guide sections](https://docs.ballerine.com/en/learn/simple_kyb_guide/)
+### Installation steps on localhost with Docker
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/installation.mdx]: Let's find a pattern for the titles we can repeat in a consistent manner. 🙂 I'm also familiar with just "Docker Usage"
```suggestion
### Local Development With Docker Steps
```

File: websites/docs/src/content/docs/en/getting_started/local_deployment.mdx
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/local_deployment.mdx:1]: Deployment guides should lean on the remote and production sides. The local parts make more sense as part of the getting started / installation guide.

File: websites/docs/src/content/docs/en/getting_started/local_development.mdx [hunk 9]
```diff
@@ -31,20 +31,3 @@ Before you can install and use Ballerine, you will need the following:
...
-
-After performing these steps, the backoffice should be running on http://localhost:5137/, the collection flow on http://localhost:5201/ and the workflow service will be accepting calls at http://localhost:3000/.
-
-The default username and password for the backoffice are:
-
-**Username:**
-<CodeBlock lang={`shell`}  code={`admin@admin.com`}/>
-
-**Password:**
-<CodeBlock lang={`shell`}  code={`admin`}/>
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/getting_started/local_development.mdx]: For us having the example as part of the "installation" page makes sense considering the get started as soon as possible logic.

File: websites/docs/astro.config.mjs [hunk 10]
```diff
@@ -38,16 +38,30 @@ export default defineConfig({
...
-              link: `/en/getting_started/deployment`,
+              label: `Ansible `,
+              link: `/en/getting_started/ansible_deployment`,
             },
-          ],
+            {
+              label: `Kubernetes Deployment`,
+              link: `/en/getting_started/kubernetes_deployment`,
+            }
+          ]
```
Comment (Omri-Levy) [inline websites/docs/astro.config.mjs]: Let's move the guides to the following;

- Learn
   - Deployment

Comment (pratapalakshmi) [reply websites/docs/astro.config.mjs]: Addressed

File: websites/docs/src/content/docs/en/deployment/ansible_deployment.mdx [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+
+## Ballerine on HTTPS
+
+### Prerequisites
+
+Before configuring SSL for your custom domain, make sure you have the following:
+
+* A domain name - You can get a custom domain from popular providers like GoDaddy, Amazon Route 53, Digital Ocean, NameCheap, and Domain.com.
+* Ports 80 and 443 are open and accessible.
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/deployment/ansible_deployment.mdx]: I think this block can be condensed and be put inline at `vite_api_url`. Something along the lines of "Incase you want to deploy ballerine on a remote server and run it on HTTPS. Note you need to own a domain and that ports 80 and 443 are forwarded".

File: websites/docs/src/content/docs/en/deployment/ansible_deployment.mdx [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+**Password:**
+<CodeBlock lang={`shell`}  code={`admin`}/>
+
+
+## Ballerine on HTTPS
+
+### Prerequisites
+
+Incase you want to deploy ballerine on a remote server and serve it on HTTPS
```
Comment (Omri-Levy) [inline websites/docs/src/content/docs/en/deployment/ansible_deployment.mdx:67]: This line is duplicated with the one in `vite_api_url`.

**Review by Omri-Levy — APPROVED** [review]

**Pull request merged**